[Bug] AppImage version fails to load remote data

### Operating System

Linux

### What's the issue you encountered?

Starting program in terminal shows following error messages:
````
[ERROR] [libimhex] Net request failed with error 77: 'Problem with the SSL CA cert (path? access rights?)'
[ERROR] [main] Init task 'Checking for updates...' threw an exception: [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal
[ERROR] [libimhex] Net request failed with error 77: 'Problem with the SSL CA cert (path? access rights?)'
[ERROR] [libimhex] Net request failed with error 77: 'Problem with the SSL CA cert (path? access rights?)'
````
Update checking doesn't work. Clicking links in "Learn" and "Help" sections does nothing, results only in error message in terminal: `*** stack smashing detected ***: terminated`

### How can the issue be reproduced?

Download latest AppImage version, set program as executable and run program.

### ImHex Version

1.21.1

### ImHex Build Type

- [ ] Nightly or built from sources

### Additional context?

Fedora Linux 35 (Workstation Edition) 64-bit

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -277,6 +277,7 @@ jobs:
             -DRUST_PATH="$HOME/.cargo/bin/"           \
             -DIMHEX_PATTERNS_PULL_MASTER=ON           \
             -DIMHEX_PLUGINS_IN_SHARE=ON               \
+            -DIMHEX_USE_BUNDLED_CA=ON                 \
             ..
           make -j 4 install DESTDIR=AppDir
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,7 @@
 cmake_minimum_required(VERSION 3.20)
 
 # Options
+option(IMHEX_USE_BUNDLED_CA "Use the CA bundle in romfs instead of the system one" OFF)
 option(IMHEX_PLUGINS_IN_SHARE "Put the plugins in share/imhex/plugins instead of lib[..]/imhex/plugins" OFF)
 option(IMHEX_STRIP_RELEASE "Strip the release builds" ON)
 option(IMHEX_OFFLINE_BUILD "Enable offline build" OFF)

--- a/cmake/build_helpers.cmake
+++ b/cmake/build_helpers.cmake
@@ -102,7 +102,11 @@ macro(detectOS)
             # Warning : Do not work with portable versions such as appimage (because the path is hardcoded)
             add_compile_definitions(SYSTEM_PLUGINS_LOCATION="${CMAKE_INSTALL_FULL_LIBDIR}/imhex") # "plugins" will be appended from the app
         endif()
-            
+
+        if(IMHEX_USE_BUNDLED_CA)
+            add_compile_definitions(IMHEX_USE_BUNDLED_CA)
+        endif()
+
     else ()
         message(FATAL_ERROR "Unknown / unsupported system!")
     endif()

--- a/cmake/build_helpers.cmake
+++ b/cmake/build_helpers.cmake
@@ -84,6 +84,8 @@ macro(detectOS)
         set(CMAKE_INSTALL_BINDIR ".")
         set(CMAKE_INSTALL_LIBDIR ".")
         set(PLUGINS_INSTALL_LOCATION "plugins")
+
+        SET(IMHEX_USE_BUNDLED_CA ON)
     elseif (APPLE)
         add_compile_definitions(OS_MACOS)
         set(CMAKE_INSTALL_BINDIR ".")
@@ -103,13 +105,14 @@ macro(detectOS)
             add_compile_definitions(SYSTEM_PLUGINS_LOCATION="${CMAKE_INSTALL_FULL_LIBDIR}/imhex") # "plugins" will be appended from the app
         endif()
 
-        if(IMHEX_USE_BUNDLED_CA)
-            add_compile_definitions(IMHEX_USE_BUNDLED_CA)
-        endif()
-
     else ()
         message(FATAL_ERROR "Unknown / unsupported system!")
     endif()
+
+    if(IMHEX_USE_BUNDLED_CA)
+        add_compile_definitions(IMHEX_USE_BUNDLED_CA)
+    endif()
+
 endmacro()
 
 # Detect 32 vs. 64 bit system

--- a/lib/libimhex/source/helpers/net.cpp
+++ b/lib/libimhex/source/helpers/net.cpp
@@ -109,7 +109,7 @@ namespace hex {
         curl_easy_setopt(this->m_ctx, CURLOPT_NOSIGNAL, 1L);
         curl_easy_setopt(this->m_ctx, CURLOPT_NOPROGRESS, 0L);
 
-#if defined(OS_WINDOWS) || defined(IMHEX_USE_BUNDLED_CA)
+#if defined(IMHEX_USE_BUNDLED_CA)
         curl_easy_setopt(this->m_ctx, CURLOPT_CAINFO, nullptr);
         curl_easy_setopt(this->m_ctx, CURLOPT_CAPATH, nullptr);
         curl_easy_setopt(this->m_ctx, CURLOPT_SSLCERTTYPE, "PEM");

--- a/lib/libimhex/source/helpers/net.cpp
+++ b/lib/libimhex/source/helpers/net.cpp
@@ -55,8 +55,9 @@ namespace hex {
         static mbedtls_x509_crt crt;
         mbedtls_x509_crt_init(&crt);
 
-        auto cacert = romfs::get("cacert.pem").string();
-        mbedtls_x509_crt_parse(&crt, reinterpret_cast<const u8 *>(cacert.data()), cacert.size());
+        std::string cacert(romfs::get("cacert.pem").string());
+        // mbedtls_x509_crt_parse needs the size of the buffer + the NULL byte provided by c_str(), hence the +1
+        mbedtls_x509_crt_parse(&crt, reinterpret_cast<const u8 *>(cacert.c_str()), cacert.size()+1);
 
         mbedtls_ssl_conf_ca_chain(cfg, &crt, nullptr);
 

--- a/lib/libimhex/source/helpers/net.cpp
+++ b/lib/libimhex/source/helpers/net.cpp
@@ -109,7 +109,7 @@ namespace hex {
         curl_easy_setopt(this->m_ctx, CURLOPT_NOSIGNAL, 1L);
         curl_easy_setopt(this->m_ctx, CURLOPT_NOPROGRESS, 0L);
 
-#if defined(OS_WINDOWS)
+#if defined(OS_WINDOWS) || defined(IMHEX_USE_BUNDLED_CA)
         curl_easy_setopt(this->m_ctx, CURLOPT_CAINFO, nullptr);
         curl_easy_setopt(this->m_ctx, CURLOPT_CAPATH, nullptr);
         curl_easy_setopt(this->m_ctx, CURLOPT_SSLCERTTYPE, "PEM");

--- a/lib/libimhex/source/helpers/net.cpp
+++ b/lib/libimhex/source/helpers/net.cpp
@@ -55,9 +55,8 @@ namespace hex {
         static mbedtls_x509_crt crt;
         mbedtls_x509_crt_init(&crt);
 
-        std::string cacert(romfs::get("cacert.pem").string());
-        // mbedtls_x509_crt_parse needs the size of the buffer + the NULL byte provided by c_str(), hence the +1
-        mbedtls_x509_crt_parse(&crt, reinterpret_cast<const u8 *>(cacert.c_str()), cacert.size()+1);
+        auto cacert = romfs::get("cacert.pem").string();
+        mbedtls_x509_crt_parse(&crt, reinterpret_cast<const u8 *>(cacert.data()), cacert.size());
 
         mbedtls_ssl_conf_ca_chain(cfg, &crt, nullptr);
 

